src/Assert.php

```diff
@@ -1710,6 +1711,8 @@ public static function subclassOf($value, $class, $message = '')
     }
 
     /**
+     * @psalm-pure
```

Comment:
`interface_exists()` cannot be pure due to it triggering autoloading in case the interface isn't already loaded 😞

Reply:
This explains why you also didn't mark `classExists` as `@pure` (my issue yesterday https://github.com/webmozart/assert/issues/185). But should it be that extreme? Consider an `@immutable` class that references some other immutable classes for the first time in the code. When it is autoloaded, those classes will also be autoloaded. Then according to your logic no class unless it's preloaded can be really considered immutable... Which is too strong for PHP, I think.

Reply:
well, I regularly work with proxy libraries (which do perform class existence checks), and something as simple as following already shows that `class_exists()` relies on global state:

```php
var_dump(class_exists(Foo::class));
eval('class Foo {}');
var_dump(class_exists(Foo::class));
```

https://3v4l.org/F4mge

So indeed: this cannot be considered pure at all, as it relies on a runtime-mutable registry of defined symbols.

Reply:
Okay, thanx. Now the PR seems to be correct :)